packages/tangle/scheduler.go

```diff
@@ -279,35 +281,50 @@ func (s *Scheduler) schedule() *Message {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	// cache the access mana
+	manaCache := s.tangle.Options.SchedulerParams.AccessManaMapRetrieverFunc()
+
+	s.updateActiveNodesList(manaCache)
+
 	start := s.buffer.Current()
 	// no messages submitted
 	if start == nil {
 		return nil
 	}
 
-	// cache the access mana retrieval
-	manas := make(map[identity.ID]float64, s.buffer.NumActiveNodes())
+	// wrap the AccessManaMap to always return at least MinMana for now
 	getCachedMana := func(id identity.ID) float64 {
-		if mana, ok := manas[id]; ok {
+		if mana, ok := manaCache[id]; ok && mana >= MinMana {
 			return mana
 		}
-		mana := math.Max(s.tangle.Options.SchedulerParams.AccessManaRetrieveFunc(id), MinMana)
-		manas[id] = mana
-		return mana
+		// always return at least MinMana
+		manaCache[id] = MinMana
+		return manaCache[id]
 	}
 
 	var schedulingNode *schedulerutils.NodeQueue
 	rounds := math.MaxInt32
 	now := clock.SyncedTime()
 	for q := start; ; {
+		nodeMana := getCachedMana(q.NodeID())
+		// clear the node from the queue, if its mana dropped below MinMana
+		if nodeMana < MinMana {
+			delete(s.deficits, q.NodeID())
+			s.buffer.RemoveNode(q.NodeID())
+			q = s.buffer.Current()
+			if q == start {
+				break
+			}
+			continue
+		}
```

Comment:
I am confused :confused: we have code that potentially removes below-min-nodes here and inside `updateActiveNodesList`.
This check here will currently never trigger, as `getCachedMana` always returns at least `MinMana`. The check in `updateActiveNodesList` could potentially trigger (since it doesn't use `getCachedMana`) but only if that node's queue is empty.
So what is the exact behavior we want during our current testing? Nodes that once had any access Mana should only be removed when their queue gets empty? And this check is there for the future case where the `MinMana` is removed?!? That might be perfectly fine, in this case we just must be aware in our tests that below-min-nodes behave differently than any other nodes.